crates/dojo-client/src/storage.rs

```diff
@@ -0,0 +1,52 @@
+use starknet::core::types::FieldElement;
+
+use crate::types::{Component, System};
+
+pub trait Storage {
+    type Error;
+
+    /// Insert a component by its name.
+    fn set_component(&mut self, name: String, component: Component) -> Result<(), Self::Error>;
```

Comment:
I think we should follow the world api as closely as possible

```suggestion
    fn register_component(&mut self, name: String, component: Component) -> Result<(), Self::Error>;
```

etc